src/integrations/terminal/TerminalProcess.ts

```diff
@@ -254,13 +254,43 @@ export class TerminalProcess extends BaseTerminalProcess {
 		// so that api request stalls to let diagnostics catch up").
 		this.stopHotTimer()
 		this.emit("completed", this.removeEscapeSequences(this.fullOutput))
+
+		// Restore focus to the webview after terminal operation completes
+		const cline = this.terminal.taskId
+			? vscode.extensions.getExtension("RooVetGit.roo-code")?.exports?.getClineProvider?.()?.getCurrentCline?.()
+			: undefined
+
+		if (cline?.provider) {
+			// Use setTimeout to ensure focus restoration happens after terminal operation is fully complete
+			setTimeout(() => {
+				cline.provider.focusWebview().catch((error: Error) => {
+					console.error("[TerminalProcess] Error restoring focus to webview:", error)
+				})
+			}, 100)
+		}
+
 		this.emit("continue")
 	}
 
 	public override continue() {
 		this.emitRemainingBufferIfListening()
 		this.isListening = false
 		this.removeAllListeners("line")
+
+		// Restore focus to the webview after terminal operation is aborted or completed
+		const cline = this.terminal.taskId
+			? vscode.extensions.getExtension("RooVetGit.roo-code")?.exports?.getClineProvider?.()?.getCurrentCline?.()
```

Comment:
This feels like a reach in terms of good separation of concerns. The terminal already fires callbacks that the `Cline` instance is listening for, and we should use that callback mechanism instead to communicate with `Cline` and `ClineProvider`. I can update this PR to demonstrate that.